// attr-button/esm2020/attr-button-group/src/attr-button-group.module.mjs
import { NgModule } from '@angular/core';
import { AttrButtonGroupComponent } from "./attr-button-group.component";
import * as i0 from "@angular/core";
export class AttrButtonGroupModule {
}
AttrButtonGroupModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: AttrButtonGroupModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AttrButtonGroupModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: AttrButtonGroupModule, declarations: [AttrButtonGroupComponent], exports: [AttrButtonGroupComponent] });
AttrButtonGroupModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: AttrButtonGroupModule, imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: AttrButtonGroupModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        AttrButtonGroupComponent
                    ],
                    imports: [],
                    exports: [
                        AttrButtonGroupComponent
                    ]
                }]
        }] });
