// attr-button/esm2020/lib/attr-button/attr-button.module.mjs
import { NgModule } from '@angular/core';
import { AttrButtonComponent } from "./attr-button.component";
import * as i0 from "@angular/core";
export class AttrButtonModule {
}
AttrButtonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: AttrButtonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AttrButtonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: AttrButtonModule, declarations: [AttrButtonComponent], exports: [AttrButtonComponent] });
AttrButtonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: AttrButtonModule, imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: AttrButtonModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        AttrButtonComponent
                    ],
                    imports: [],
                    exports: [
                        AttrButtonComponent
                    ]
                }]
        }] });
